### ISSUE TITLE
Module not found: Error: Can't resolve 'jquery'  in webpack

### ISSUE THREAD

**Issue by oshihirii** [fengyuanchen/datepicker#153]
When running webpack, I get this error:

> ERROR in ./src/js/datepicker.js
> Module not found: Error: Can't resolve 'jquery' in 'C:\Users\Me\Documents\my_repo\src\js'
>  @ ./src/js/datepicker.js 16:124-141 16:190-217

common.js:

    import datepicker from './datepicker';

webpack.config.js:

```
module.exports = {
    entry: "./src/js/common.js",
    output: {
        filename: "[name].js",
        path: path.resolve(__dirname, "dist/js")
    },
```

 Datepicker v0.6.5

Does anyone know how to fix error?

**Comment by automagisch** [comment]
You need to install `jQuery` in order for this package to work, have you installed it?

**Comment by oshihirii** [comment]
Yes, I have, like this:

    import {jQuery as $} from './jquery';
    import datepicker from './datepicker';

I also tried various combinations of the jQuery import, as shown in this post:

https://stackoverflow.com/questions/34338411/how-to-import-jquery-using-es6-syntax

    import * as $ from './jquery';
    
    import {$, jQuery} from './jquery';
    
    import {$,jQuery} from './jquery';
    // export for others scripts to use
    window.$ = $;
    window.jQuery = jQuery;

**Comment by fengyuanchen** [comment]
```js
import $ from 'jquery';
import '@fengyuanchen/datepicker/dist/datepicker.esm.js';
import '@fengyuanchen/datepicker/dist/datepicker.css';

$('#myInput').datepicker();
```

**Comment by oshihirii** [comment]
**Update**

I changed:

    import $ from 'jquery';

to:

   import $ from './jquery';

in `datepicker.esm` and the build seems to work.

**Comment by andrewfinnell** [comment]
I'm running into this same issue with another common package that I'm building with webpack. I clearly have 'jquery' available, but for some reason it cannot be found. Was this resolved without having to change the third-party import?

**Comment by flameoftheforest** [comment]
This solution works for me:
https://stackoverflow.com/questions/45330231/webpack-cant-resolve-jquery

tldr;:
include:
```
externals: {
    // require("jquery") is external and available
    //  on the global var jQuery
    "jquery": "jQuery"
  }
```
in `webpack.config.js`

**Comment by aesyondu** [comment]
> This solution works for me:
> https://stackoverflow.com/questions/45330231/webpack-cant-resolve-jquery
> 
> tldr;:
> include:
> 
> ```
> externals: {
>     // require("jquery") is external and available
>     //  on the global var jQuery
>     "jquery": "jQuery"
>   }
> ```
> 
> in `webpack.config.js`

Thanks it worked. More info here https://webpack.js.org/configuration/externals/

**Comment by Alepandis** [comment]
Hi, the message recommendation is in that run de command npm install --save jquery, i run this command and the error was fixed, hope someone helps :)

**Comment by cassiohalmeida** [comment]
In case you're using Laravel Mix and have the same problem:

On your `webpack.mix.js` file

Just below

```
let mix = require('laravel-mix');
Add the following code

mix.webpackConfig({
    externals: {
        "jquery": "jQuery"
    }
});
```
Add any other external as required. For example I decided to load external React and ReactDOM so my config is

```
mix.webpackConfig({
    externals: {
        "react": "React",
        "react-dom": "ReactDOM"
    }
});
```

Font: [https://stackoverflow.com/questions/43721132/setup-external-libraries-with-laravel-mix](https://stackoverflow.com/questions/43721132/setup-external-libraries-with-laravel-mix)

**Comment by nnbaokhang** [comment]
> Hi, the message recommendation is that run de command npm install --save jquery, i run this command and the error was fixed, hope someone helps :)

This works for me
